plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/HiveS3Config.java

```diff
@@ -31,6 +31,7 @@
 
 import java.io.File;
 import java.util.List;
+import java.util.Optional;
```

Comment:
Change the commit message to `Add configuration for the expiration time of AWS S3 connections`

The fact that the number of exceptions will decrease for high-load situations in which  Trino user has this setting at hand is a side effect.


Maybe also add in the commit message the quote from https://github.com/trinodb/trino/issues/16005

```
When a certain HTTP connection has been idle for a period of time , S3 will close the connection, but the SDK later tries to reuse the connection, then the Unable to execute HTTP request.
```

Potentially off-topic: Why doesn't this problem occur more often in other Trino installations? It's not like every Trino server does continuosly perform S3 queries. I find it a rather normal scenario to have more load in the morning when people come to work, but haven't heard about this issue from other `trino-hive` users querying on top of AWS S3.

Reply:
> Change the commit message to `Add configuration for the expiration time of AWS S3 connections`
> 
> The fact that the number of exceptions will decrease for high-load situations in which Trino user has this setting at hand is a side effect.
> 
> Maybe also add in the commit message the quote from #16005
> 
> ```
> When a certain HTTP connection has been idle for a period of time , S3 will close the connection, but the SDK later tries to reuse the connection, then the Unable to execute HTTP request.
> ```
> 
> Potentially off-topic: Why doesn't this problem occur more often in other Trino installations? It's not like every Trino server does continuosly perform S3 queries. I find it a rather normal scenario to have more load in the morning when people come to work, but haven't heard about this issue from other `trino-hive` users querying on top of AWS S3.

@findinpath Thank you for your advice very much! I have improved several aspects you mentioned above, please review again when you have time～  

Yes the problem didn't happen in all of our Trino clusters(each cluster has a different network and cloud environment), so when i saw the advice from AWS support team, i realized that it is likely because the network instability from the company intranet proxies to the same public network gateway, and the high load in the morning just exacerbated the effect.

But just as the email from AWS said, the external Internet factors are difficult to control, some users don't want the queries to fail due to this kind of service jitter, which is not good for user experience and robustness in some BI report scenarios.  

So we want to give other users a choice to decide whether to adjust this.